program/src/processor.rs

```diff
@@ -572,6 +610,10 @@ impl Processor {
         // this is somewhat subtle, but if the stake account is Uninitialized, you pass
         // it twice and sign ie, Initialized or Stake, we use real withdraw
         // authority. Uninitialized, stake account is its own authority
+        // JC: do you actually need to pass it twice? I imagine in the current
+        // version, you could just pass it once as a signer and it would Just Work,
+        // but this falls in the same camp as the other similar comments: if
+        // people are depending on that current weird behavior, we break 'em.
```

Comment:
youre right, i was treating the doc comment on the withdraw instruction literally rather than seriously. they get the account by index off the invoke context in the existing stake program, which i assume means its deduped to the highest allowed privilege

in fact the nature of the transaction format means that its impossible to restrict privlege on the instruction level, doesnt it? the transaction defines what accounts are writable or signers, and each instruction account only had an index into the accounts list. including the same account multiple times with different privileges actually escalates them all to the highest. we dont need to accept the account multiple times at all

Reply:
Yep that's exactly correct